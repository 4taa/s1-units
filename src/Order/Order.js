import React from 'react';
import './Order.css';

export default class Order extends React.PureComponent {
	getItems() {
		const {order: {items}} = this.props;

		if (!items || !items.length) {
			return null;
		}

		return items.map((item, index) => (
			<div
				key={index}
				className='Order-item'
			>
				{item}
			</div>
		))
	}

	render() {
		const {order: {shop, date}} = this.props;

		//напишем getDate как в letter.meta
		return (
			<div className='Order'>
				<div className='Order-header'>
					<span className='Order-shop'>{shop}</span>
					<span>{date}</span>
				</div>

				<div className='Order-items'>
					{this.getItems()}
				</div>
			</div>
		)
	}
}
