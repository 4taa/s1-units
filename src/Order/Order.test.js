import { shallow, configure } from 'enzyme';
import React from 'react';

import {fakeOrders} from '../mock/fakeOrders';
import Order from './Order';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('Order component', () => {
    const wrapper1 = shallow(<Order order={fakeOrders[0]}/>);
    const wrapper2 = shallow(<Order order={fakeOrders[1]}/>);
    const wrapper3 = shallow(<Order order={fakeOrders[2]}/>);
    const wrapperNull = shallow(<Order order={{
		id: 123,
		date: 1544356800000,
		shop: 'Ali Express',
		items: []
    }}/>);
    
    it('render first', () => {
        expect(toJson(wrapper1)).toMatchSnapshot();
    })

    it('render second', () => {
        expect(toJson(wrapper2)).toMatchSnapshot();
    })

    it('render third', () => {
        expect(toJson(wrapper3)).toMatchSnapshot();
    })

    it('render none', () => {
        expect(toJson(wrapperNull)).not.toEqual(toJson(wrapper1));
    })
})