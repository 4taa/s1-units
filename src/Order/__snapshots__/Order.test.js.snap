// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order component render first 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Ali Express
    </span>
    <span>
      9 декабря, вс, 2018 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;

exports[`Order component render second 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Lamoda.ru
    </span>
    <span>
      13 марта, ср, 2019 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Жакет - BOREAL5
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Miss Gabby Костюм
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Ostin перчатки мужские
    </div>
    <div
      className="Order-item"
      key="3"
    >
      Zara худи роз.
    </div>
  </div>
</div>
`;

exports[`Order component render third 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Эльдорадо
    </span>
    <span>
      14 марта, чт, 2019 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Ноутбук Apple MacBook Air 13.3" (MQD32RU/A)
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Игровая приставка Sony PlayStation 4 Pro 1TB Black (CUH-7208B)
    </div>
  </div>
</div>
`;
